src/converse-chatview.js

```diff
@@ -1312,6 +1312,10 @@ converse.plugins.add('converse-chatview', {
                 if (!views.get(item.get('id')) && item.get('type') === _converse.PRIVATE_CHAT_TYPE) {
                     views.add(item.get('id'), new _converse.ChatBoxView({model: item}));
                 }
+                const servive_administration_chat = _converse.chatboxviews.el.querySelector('#sendAdminCommandId');
+                if(!_.isNil(servive_administration_chat)) {
+                    servive_administration_chat.remove();
+                }
```

Comment:
This looks wrong. The chatview plugin does not have a dependency on the service-admin plugin, so it should not have to be aware of anything related to that plugin and therefore this code should not be here.

Looks like you're adding the `#sendAdminCommandId` element to *all* chat boxes and then removing it again here, right? You shouldn't be adding it to all of them in the first place, only to the relevant one.

Reply:
Thanks for the feedback. This one is a litte more complicated. That's the problem i refered to when I wrote 

> I also tried to add the views as a "chat" so that converse can handle the visibility automatically. But this attempt failed :( So now I added code to clean the view manually.

The problem here (mainly in fullscreen-mode) is that the service-administration-window (left on the screenshot) is not a real chat. So when the user opens a normal chat/muc I have to manually close the service-administration-windows so that the chat will be displayed. Normally converse.js does this by hiding the current chat and then opening the new chat (in fullscreen). 

The best way would be if the service-administration-window would be handled like the other chats. That way no manual cleanup would be required. That way the code inside `converse-controlbox.js` could also be removed.

I assume that the new view has to be added to `_converse.chatboxviews` and `_converse.chatboxes`. This way converse can handle displaying/removing the view. But I couldn't get it to work. That's why I clean it manually here, when a new chat is opened. I could try to create my own event-listeners inside `converse-service-administration.js` to clean the view. That way the code wouldn't pollute the other plugins.

Reply:
> The best way would be if the service-administration-window would be handled like the other chats.

Take a look at the `HeadlinesBox` and `HeadlinesBoxView` in [converse-headline.js](https://github.com/conversejs/converse.js/blob/master/src/converse-headline.js).

That's also not a "real chat" but they both extend the chatbox base classes and therefore don't have this problem. You should be able to do the same with the service administration box. I think it's conceptually very similar. Same with the `ControlBoxView`, it also extends the `ChatBoxView` baseclass.

Currently the headlines box doesn't appear in fullscreen but that's more a UX issue than anything else. There's nowhere to click to bring it to the foreground :D
You don't have this problem because you added a section in the controlbox for opening the service admin box.

I'd like to someday refactor the base classes so that the one-on-one and MUC boxes both extend a common base class instead of everything extending the current one-on-one class, but I don't have time for it currently and it isn't (yet) really an issue.